pkg/controller/plan/migration.go

```diff
@@ -593,31 +593,37 @@ func (r *Migration) execute(vm *plan.VMStatus) (err error) {
 			vm.AddError(fmt.Sprintf("Step '%s' not found", r.step(vm)))
 			break
 		}
+		var pvcNames []string
 		var ready bool
 		ready, err = r.builder.BeforeTransferHook(r.provider, vm.Ref)
 		if err != nil {
 			err = liberr.Wrap(err)
 			return
 		}
-		if !ready {
-			r.Log.Info("BeforeTransfer hook isn't ready yet")
-			return
-		}
-		var pvcNames []string
+
 		if r.kubevirt.isOpenstack(vm) {
-			pvcNames, err = r.kubevirt.createOpenStackVolumes(vm.Ref)
-			if err != nil {
-				step.AddError(err.Error())
-				err = nil
+			pvcNames, err = r.kubevirt.ensureOpenStackVolumes(vm.Ref, ready)
+			if !ready {
 				return
 			}
-			err = r.kubevirt.createPodToBindPVCs(vm, pvcNames)
 			if err != nil {
 				step.AddError(err.Error())
 				err = nil
 				break
 			}
 		}
+
+		if !ready {
+			r.Log.Info("BeforeTransfer hook isn't ready yet")
+			return
+		}
+
+		err = r.kubevirt.createPodToBindPVCs(vm, pvcNames)
+		if err != nil {
+			step.AddError(err.Error())
+			err = nil
+			break
+		}
```

Comment:
maybe it's a result of rebase conflict - it seems incorrect, did you intentionally pushed it out from the `if r.kubevirt.isOpenstack(vm)` scope?

Reply:
yeah, it's not intentional, I forgot to wrap it again